### PR TITLE
Fix race with Jet activation

### LINKED ISSUE #19183: Race between starting Jet and activating it
In `JetExtension`, the `activated` field is set in `afterStart`. Reckoning from the name, it happens some time after the start. But in the meantime, a job can be submitted and it fails with `Jet is disabled because the current cluster version is less than 5.0` - a very misleading error because I started only 5.0 members.

### PR DESCRIPTION
**Pull request by viliam-durina**
Fixes #19183

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java [hunk 1]
```diff
@@ -70,18 +67,34 @@ public void beforeStart() {
     }
 
     public void afterStart() {
-        if (node.isRunning() && node.getClusterService().getClusterVersion().isGreaterOrEqual(Versions.V5_0)) {
-            activated = true;
-            jetServiceBackend.getJobCoordinationService().startScanningForJobs();
+        if (tryActivate()) {
             logger.info("Jet is enabled");
```
Comment (ufukyilmaz) [inline hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java:71]: I think we should move this "Jet is enabled" log into `tryActivate`'s activation block.

**Review by ufukyilmaz — APPROVED** [review]
It looks good to me.

File: hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java [hunk 2]
```diff
@@ -95,24 +108,22 @@ public void beforeClusterStateChange(ClusterState requestedState) {
...
-        if (!activated && isAfterStartCalled && newVersion.isGreaterOrEqual(Versions.V5_0)) {
-            // Activate Jet after rolling upgrade in which the cluster
-            // version is upgraded from 4.x to 5.0
-            activated = true;
-            jetServiceBackend.getJobCoordinationService().startScanningForJobs();
+    public void onClusterVersionChange() {
+        // Activate Jet after rolling upgrade in which the cluster
+        // version is upgraded from 4.x to 5.0
+        if (!activated.get() && tryActivate()) {
             logger.info("Jet is enabled after the cluster version upgrade.");
```
Comment (ufukyilmaz) [inline hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java:121]: But, this log can be missed when jet is activated after the cluster version upgrade and this activation is performed on another place such as on getJet.

**Review by ufukyilmaz — CHANGES_REQUESTED** [review]
Because the `Jet is enabled after the cluster version upgrade.` log printed in `onClusterVersionChange` may be lost, I wanted to revert my approval

**Review by ufukyilmaz — APPROVED** [review]
LGTM

**Review by gurbuzali — APPROVED** [review]

**Pull request merged**